examples/main.rs

```diff
@@ -39,41 +40,42 @@ impl backend::DataQueryError for QueryError {
     }
 }
 
-type QueryDataIter = std::iter::Map<
-    std::vec::IntoIter<backend::DataQuery>,
-    fn(backend::DataQuery) -> Result<backend::DataResponse, QueryError>,
->;
-
 #[tonic::async_trait]
 impl backend::DataService for MyPluginService {
     type QueryError = QueryError;
-    type Iter = QueryDataIter;
-    async fn query_data(&self, request: backend::QueryDataRequest) -> Self::Iter {
-        request.queries.into_iter().map(|x| {
-            // Here we create a single response Frame for each query.
-            // Frames can be created from iterators of fields using [`IntoFrame`].
-            Ok(backend::DataResponse::new(
-                x.ref_id.clone(),
-                vec![[
-                    // Fields can be created from iterators of a variety of
-                    // relevant datatypes.
-                    [
-                        Utc.ymd(2021, 1, 1).and_hms(12, 0, 0),
-                        Utc.ymd(2021, 1, 1).and_hms(12, 0, 1),
-                        Utc.ymd(2021, 1, 1).and_hms(12, 0, 2),
-                    ]
-                    .into_field("time"),
-                    [1_u32, 2, 3].into_field("x"),
-                    ["a", "b", "c"].into_field("y"),
-                ]
-                .into_frame("foo")
-                .check()
-                .map_err(|source| QueryError {
-                    ref_id: x.ref_id,
-                    source,
-                })?],
-            ))
-        })
+    type Stream = backend::BoxDataResponseStream<Self::QueryError>;
+    async fn query_data(&self, request: backend::QueryDataRequest) -> Self::Stream {
+        Box::pin(
```

Comment:
Out of curiosity why does it have to be pinned?

Reply:
Hmm, I think it _might_ be possible to just require it to be `Unpin` instead but my knowledge of pinning is a bit fuzzy... The [tokio docs](https://tokio.rs/tokio/tutorial/streams#iteration) explain that 'calling `next()` on a stream requires the stream to be pinned`, so we need the user to return a `Pin<Box<...>>` (or maybe(?!?) add the `Unpin` trait bound to the associated type?). I've basically just replicated the [`tonic` examples](https://github.com/hyperium/tonic/blob/01e5be508051eebf19c233d48b57797a17331383/examples/src/routeguide/server.rs#L105) though 😄